dfastmi/io/ConfigFileOperations.py

```diff
@@ -47,20 +55,211 @@ def write_config(filename: str, config: configparser.ConfigParser) -> None:
         """
         sections = config.sections()
         ml = 0
-        for s in sections:
-            options = config.options(s)
+        for section in sections:
+            options = config.options(section)
             if len(options) > 0:
                 ml = max(ml, max([len(x) for x in options]))
 
         OPTIONLINE = "  {{:{}s}} = {{}}\n".format(ml)
-        with open(filename, "w") as configfile:
+        with Path(filename).open("w", encoding="utf-8") as configfile:
             first = True
-            for s in sections:
+            for section in sections:
                 if first:
                     first = False
                 else:
                     configfile.write("\n")
-                configfile.write("[{}]\n".format(s))
-                options = config.options(s)
+                configfile.write(f"[{section}]\n")
+                options = config.options(section)
                 for o in options:
-                    configfile.write(OPTIONLINE.format(o, config[s][o]))
+                    configfile.write(OPTIONLINE.format(o, config[section][o]))
+
+    @staticmethod
+    def save_configuration_file(filename: str, config):
```

Comment:
A `ConfigParser` object is passed to a writing function, this does not seem correct to me, as I would expect a parser to parse a configuration file.

Reply:
Not in scope of this issue: DFAST-108